Mathlib/CategoryTheory/Triangulated/Opposite/Functor.lean

```diff
@@ -0,0 +1,208 @@
+/-
+Copyright (c) 2024 Sophie Morel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sophie Morel
+-/
+import Mathlib.CategoryTheory.Triangulated.Opposite.Pretriangulated
+
+/-!
+# Opposites of functors between pretriangulated categories,
+
+If `F : C ⥤ D` is a functor between pretriangulated categories, we prove that
+`F` is a triangulated functor if and only if `F.op` is a triangulated functor.
+In order to do this, we first show that a `CommShift` structure on `F` naturally
+gives one on `F.op` (for the shifts on `Cᵒᵖ` and `Dᵒᵖ` defined in
+`CategoryTheory.Triangulated.Opposite.Basic`), and we then prove
+that `F.mapTriangle.op` and `F.op.mapTriangle` correspond to each other via the
+equivalences `(Triangle C)ᵒᵖ ≌ Triangle Cᵒᵖ` and `(Triangle D)ᵒᵖ ≌ Triangle Dᵒᵖ`
+given by `CategoryTheory.Pretriangulated.triangleOpEquivalence`.
+
+-/
+
+namespace CategoryTheory
+
+namespace Functor
+
+open Category Limits Pretriangulated Opposite
+
+variable {C D : Type*} [Category C] [Category D] [HasShift C ℤ] [HasShift D ℤ]
+
+variable (F : C ⥤ D) [F.CommShift ℤ]
+
+/-- If `F` commutes with shifts, so does `F.op`, for the shifts chosen on `Cᵒᵖ` in
+`CategoryTheory.Triangulated.Opposite.Basic`.
+-/
+noncomputable def commShiftOpInt : @Functor.CommShift Cᵒᵖ Dᵒᵖ _ _ F.op ℤ _
```

Comment:
The file should start mostly as follows. The definition of the `CommShift` should be in the `Pretriangulated.Opposite` namespace so that it becomes automatically available when we `open Pretriangulated.Opposite`:
```lean
namespace CategoryTheory

variable {C D : Type*} [Category C] [Category D] [HasShift C ℤ] [HasShift D ℤ]
  (F : C ⥤ D) [F.CommShift ℤ]

namespace Pretriangulated.Opposite

noncomputable scoped instance commShiftOpInt : F.op.CommShift ℤ := by
  letI F' : OppositeShift C ℤ ⥤ OppositeShift D ℤ := F.op
  letI : F'.CommShift ℤ := F.commShiftOp ℤ
  apply F'.commShiftPullback

end Pretriangulated.Opposite

namespace Functor

noncomputable example : F.op.CommShift ℤ := inferInstance -- fails

open Pretriangulated.Opposite -- from now on, we have the shift on `Cᵒᵖ` and on `F.op`

noncomputable example : F.op.CommShift ℤ := inferInstance -- succeeds
```
Then, `letI` should no longer be necessary in the code below.

Reply:
Wasn't sure how to manage the `CommShift` instance on `F.op`. I'll try this.